### PR TITLE
Fix adjoint spring damper element

### PR DESCRIPTION
**Pull request by MFusseder**
The proposed modifications solve the problem reported by @philbucher  when running the test case of the adjoint spring damper element in `FullDebug`.

### CODE REVIEW

File: applications/StructuralMechanicsApplication/custom_response_functions/adjoint_elements/adjoint_finite_difference_spring_damper_element_3D2N.cpp [hunk 1]
```diff
@@ -28,8 +28,19 @@ void AdjointFiniteDifferenceSpringDamperElement<TPrimalElement>::InitializeSolut
...
+        const auto temp = this->GetValue(NODAL_DISPLACEMENT_STIFFNESS);
+        this->pGetPrimalElement()->SetValue(NODAL_DISPLACEMENT_STIFFNESS, temp);
+        stiffness_available = true;
+    }
+    if (this->Has(NODAL_ROTATIONAL_STIFFNESS)) {
+        const auto temp = this->GetValue(NODAL_ROTATIONAL_STIFFNESS);
+        this->pGetPrimalElement()->SetValue(NODAL_ROTATIONAL_STIFFNESS, temp);
+        stiffness_available = true;
+    }
+    KRATOS_ERROR_IF_NOT(stiffness_available) << "Neither NODAL_DISPLACEMENT_STIFFNESS nor NODAL_ROTATIONAL_STIFFNESS available!" << std::endl;
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_response_functions/adjoint_elements/adjoint_finite_difference_spring_damper_element_3D2N.cpp]: I would do this check in `Check`

File: applications/StructuralMechanicsApplication/custom_response_functions/adjoint_elements/adjoint_finite_difference_spring_damper_element_3D2N.cpp [hunk 2]
```diff
@@ -28,8 +28,19 @@ void AdjointFiniteDifferenceSpringDamperElement<TPrimalElement>::InitializeSolut
...
-    this->pGetPrimalElement()->SetValue(NODAL_ROTATIONAL_STIFFNESS, this->GetValue(NODAL_ROTATIONAL_STIFFNESS));
+    auto stiffness_available = false;
+    if (this->Has(NODAL_DISPLACEMENT_STIFFNESS)) {
+        const auto temp = this->GetValue(NODAL_DISPLACEMENT_STIFFNESS);
+        this->pGetPrimalElement()->SetValue(NODAL_DISPLACEMENT_STIFFNESS, temp);
+        stiffness_available = true;
+    }
+    if (this->Has(NODAL_ROTATIONAL_STIFFNESS)) {
+        const auto temp = this->GetValue(NODAL_ROTATIONAL_STIFFNESS);
+        this->pGetPrimalElement()->SetValue(NODAL_ROTATIONAL_STIFFNESS, temp);
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_response_functions/adjoint_elements/adjoint_finite_difference_spring_damper_element_3D2N.cpp]: I would create a `const&` of the adjoint element and then set it in the primal
like this:
~~~c++
const auto& r_const_this = *this;

this->pGetPrimalElement()->SetValue(NODAL_ROTATIONAL_STIFFNESS, r_const_this.GetValue(NODAL_ROTATIONAL_STIFFNESS));

~~~

**Review by philbucher — COMMENTED** [review]
I think you can improve your code with using `const&` to the adjoint element

have a look at my suggestions, then I can explain if necessary

File: applications/StructuralMechanicsApplication/custom_response_functions/adjoint_elements/adjoint_finite_difference_spring_damper_element_3D2N.cpp (hunk 2 above)
Comment (MFusseder) [reply applications/StructuralMechanicsApplication/custom_response_functions/adjoint_elements/adjoint_finite_difference_spring_damper_element_3D2N.cpp]: Thank you for the suggestion. I will try it.

File: applications/StructuralMechanicsApplication/custom_response_functions/adjoint_elements/adjoint_finite_difference_spring_damper_element_3D2N.cpp (hunk 1 above)
Comment (MFusseder) [reply applications/StructuralMechanicsApplication/custom_response_functions/adjoint_elements/adjoint_finite_difference_spring_damper_element_3D2N.cpp]: I don't think that is possible. The stiffness values typically set to the elements by `AssignVectorVariableToEntitiesProcess` which executes within `InitializeSolutionStep` to my knowledge.

Comment (philbucher) [reply applications/StructuralMechanicsApplication/custom_response_functions/adjoint_elements/adjoint_finite_difference_spring_damper_element_3D2N.cpp]: ok then I would do the following: 

~~~c++
const auto& r_const_this = *this;
this->pGetPrimalElement()->SetValue(NODAL_DISPLACEMENT_STIFFNESS, r_const_this.GetValue(NODAL_DISPLACEMENT_STIFFNESS));
this->pGetPrimalElement()->SetValue(NODAL_ROTATIONAL_STIFFNESS, r_const_this.GetValue(NODAL_ROTATIONAL_STIFFNESS));

KRATOS_ERROR_IF_NOT(this->Has(NODAL_DISPLACEMENT_STIFFNESS) || this->Has(NODAL_ROTATIONAL_STIFFNESS))
~~~

Comment (MFusseder) [reply applications/StructuralMechanicsApplication/custom_response_functions/adjoint_elements/adjoint_finite_difference_spring_damper_element_3D2N.cpp]: you are right. I modified it as proposed.

**Review by philbucher — APPROVED** [review]
:+1: thanks

**Pull request merged**